core-management/core-management-subsystem/src/main/java/org/wildfly/extension/core/management/ProcessStateListenerResourceDefinition.java

```diff
@@ -0,0 +1,154 @@
+            .setRestartAllServices()
+            .build();
+    public static final AttributeDefinition LISTENER_MODULE = SimpleAttributeDefinitionBuilder.create(MODULE, ModelType.STRING, false)
+            .setAllowExpression(false)
+            .setRestartAllServices()
+            .build();
+    public static final AttributeDefinition TIMEOUT = SimpleAttributeDefinitionBuilder.create(ModelDescriptionConstants.TIMEOUT, ModelType.INT, true)
+            .setAllowExpression(true)
+            .setDefaultValue(new ModelNode(30))
+            .setMeasurementUnit(MeasurementUnit.SECONDS)
+            .setRestartAllServices()
+            .build();
+
+    @Override
+    public void registerChildren(ManagementResourceRegistration resourceRegistration) {
+        super.registerChildren(resourceRegistration);
+    }
+
+    private static final AttributeDefinition[] ATTRIBUTES = new AttributeDefinition[] {
+            LISTENER_CLASS, LISTENER_MODULE, PROPERTIES, TIMEOUT
+    };
+
+    ProcessStateListenerResourceDefinition() {
+        super(new Parameters(PROCESS_STATE_LISTENER_PATH, CoreManagementExtension.getResourceDescriptionResolver("process-state-listener"))
+                .setOrderedChild()
+                .setAddHandler(new ProcessStateListenerResourceDefinition.ProcessStateListenerAddHandler())
+                .setRemoveHandler(new ProcessStateListenerResourceDefinition.ProcessStateListenerRemoveHandler()));
+    }
+
+    @Override
+    public Collection<AttributeDefinition> getAttributes() {
+        return Arrays.asList(ATTRIBUTES);
+    }
+
+    private static class ProcessStateListenerAddHandler extends AbstractAddStepHandler  {
+
+        ProcessStateListenerAddHandler() {
+            super(PROCESS_STATE_LISTENER_CAPABILITY, ATTRIBUTES);
+        }
+
+        @Override
+        protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
+            super.performRuntime(context, operation, model);
+            String className = LISTENER_CLASS.resolveModelAttribute(context, model).asString();
+            String moduleIdentifier = LISTENER_MODULE.resolveModelAttribute(context, model).asString();
+            ProcessStateListener listener = newInstance(className, moduleIdentifier);
+            Map<String, String> properties = PROPERTIES.unwrap(context, model);
+            int timeout = TIMEOUT.resolveModelAttribute(context, model).asInt();
+
+            ProcessStateListenerService.install(context.getServiceTarget(),
+                    context.getProcessType(),
+                    context.getRunningMode(), context.getCurrentAddress().getLastElement().getValue(),
+                    listener,
+                    properties,
+                    timeout);
+        }
+
+        @Override
+        protected boolean requiresRuntime(OperationContext context) {
+            return super.requiresRuntime(context) || context.getRunningMode() == RunningMode.ADMIN_ONLY;
```

Comment:
Probably just return true
* if processType.isServer()
* Otherwise, check that the address begins with the 'host' and 'subsystem' keys